### PR TITLE
Migration of sessions in KC25 should run only on migration, not on imports

### LINKED ISSUE #30351: Migration of sessions in KC25 should run only on migration, not on imports
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

The migration is triggered also on import

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

It should only run on the migration of a Keycloak installation

### Actual behavior

It runs also on imports, when the import is from an older version of Keycloak

### How to Reproduce?

Occurred in the test suite when testing things for KC26

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ahus1**
Closes #30351

### CODE REVIEW

**Comment by ahus1** [comment]
@mhajas - The tests are green, and I tested locally that the migration is called upon migration when a new version of Keycloak starts up. 

Please give it a review when you have the time. I plan to do a backport to KC25 once this is merged. Thanks!

**Review by mhajas — APPROVED** [review]
Thank you for the fix. It makes sense to me, however, I am not sure it makes sense to merge this fix to main, since this code will be removed in 26. We can send the PR only to 25. But still I am approving and leave it to you whether we want to merge or no.

**Comment by ahus1** [comment]
I'm so used to first do things in upstream, and then backport them, let's stick to that. :-)

**Pull request merged**
